test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2528,9 +2529,10 @@ public static <T> T safeGet(Future<T> future) {
      *
      * @return The value with which the {@code future} was completed.
      */
-    public static <T> T safeGet(Future<T> future, long timeout, TimeUnit unit) {
+    // NB private because tests should be designed not to need to wait for longer than SAFE_AWAIT_TIMEOUT.
+    private static <T> T safeGet(Future<T> future, TimeValue timeout) {
```

Comment:
Making this private means we cannot have a different timeout for awaitIndexExists in #126501. It should be OK except maybe the one used by remote cluster client.